pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2487,3 +2487,129 @@ def test_mergeerror_on_left_index_mismatched_dtypes():
     df_2 = DataFrame(data=["X"], columns=["C"], index=[999])
     with pytest.raises(MergeError, match="Can only pass argument"):
         merge(df_1, df_2, on=["C"], left_index=True)
+
+
+@pytest.mark.parametrize(
```

Comment:
Following things are missing on first glance:

- joining over more than one column
- joining with something like left_index=True, right_on="X"
- merge finds common colls to join over if nothing is provided
- check that nan is matched with nan to be consistent
- what happens with different dtypes?

Reply:
your tests are very specific and you might miss some of the cases @pholf discusses. You might try multiple `parametrizations` although not too many since you don't want combinations to explode.
e.g.

```python
@pytest.mark.parametrize("left", [{"left_on": "A"}, {"left_index": True}])
@pytest.mark.parametrize("right", [{"right_on": "B"}, {"right_index": True}])
...
kwargs = {**left, **right, **others}
result = merge(**kwargs)
...
```

You may to design a clever routine for expected output based on the different combinations.

Reply:
Hi @phofl , @attack68 .. I will work on these missing cases. Little difficult to parametrize like that (as the output will be different). Maybe I will setup multiple tests to check specific cases.

Reply:
Yeah for sure, The outputs vary too much to geht them with one test I think

Reply:
Hi @phofl @attack68 - I think I have covered all the cases mentioned above. Let me know how this looks.